src/component/project/ProjectAccess/ProjectAccess.tsx

```diff
@@ -78,58 +63,35 @@ const ProjectAccess = () => {
         );
     }
 
-    const handleRoleChange = (userId, currRoleId) => async evt => {
-        const roleId = evt.target.value;
-        try {
-            await projectApi.removeUserFromRole(id, currRoleId, userId);
-            await projectApi.addUserToRole(id, roleId, userId).then(() => {
+    const handleRoleChange =
+        (userId: string, currRoleId: string) => async (evt: React.ChangeEvent<HTMLInputElement>) => {
+            const roleId = evt.currentTarget.value;
+            try {
+                await removeUserFromRole(id, currRoleId, userId);
+                await addUserToRole(id, roleId, userId);
+                refetchProjectAccess();
+
                 setToastData({
                     type: 'success',
                     title: 'User role changed successfully',
                 });
-            });
-            const newUsers = users.map(u => {
-                if (u.id === userId) {
-                    return { ...u, roleId };
-                } else return u;
-            });
-            setUsers(newUsers);
-        } catch (err) {
-            setToastData({
-                type: 'error',
-                title: err.message || 'Server problems when adding users.',
-            });
-        }
-    };
-
-    const addUser = async (userId, roleId) => {
-        try {
-            await projectApi.addUserToRole(id, roleId, userId);
-            await fetchAccess().then(() => {
+            } catch (err: any) {
                 setToastData({
-                    type: 'success',
-                    title: 'Successfully added user to the project',
+                    type: 'error',
+                    title: err.message || 'Server problems when adding users.',
                 });
-            });
-        } catch (err) {
-            setToastData({
-                type: 'error',
-                title: err.message || 'Server problems when adding users.',
-            });
-        }
-    };
+            }
+        };
 
-    const removeAccess = (userId: number, roleId: number) => async () => {
+    const removeAccess = (userId: string, roleId: string) => async () => {
```

Comment:
I generally find it a bit strange that useId and roleId are strings when I "know" they are numbers inside unleash. But it should not matter what they are from the frontend perspective, so maybe strings are better.

Reply:
I agree. We should try to keep it as consistent as possible.